### ISSUE TITLE
Add an option to copy and paste widget configuration settings

### ISSUE THREAD

**Issue by ponceta** [qgis/QGIS#58971]
### Feature description

Inspired by --- copy / paste symbol --- copy / paste style :

 - [ ] Add an option to copy and paste widget configuration settings

![image](https://github.com/user-attachments/assets/c62546b6-e671-478d-b7c6-8fe2c60acf56)

Scope : 
- [ ] ALL sections
- [ ] display
- [ ] general
- [ ] widget type
- [ ] constraints
- [ ] default values
- [ ] strategy

Avoid :

- [ ] Alias
- [ ] Comment

Exceptions :

- [ ] Field does not accept such widget / configuration

### Additional context

Widget configurations are valued with more and more options, we should ease the maintenance of these configuration via copy paste tools like we did a few years ago with copy paste style.

### Uses cases 

1. Between similar fields configuration (dates, value lists, Not NULL values, ...)
(2. Between projects)

**Comment by ponceta** [comment]
@3nids does it sounds like a good idea? Feasible or with a huge complexity?

**Comment by andreasneumann** [comment]
@ponceta - are you thinking about all widgets together for the whole layer (I think this is already possible) or one or more selected individual widgets?

Probably the latter?

**Comment by ponceta** [comment]
My particular exemple is that one widget configuration is the same for several fields and I have to make exactly the same clics and configuration so my widgets looks the same. It is then easy to miss a click and it is very time consuming.

Here i have fk_dataowner, fk_provider, fk_owner, fk_operator all referincing an organisation table and having exactly the same widget configuration. And this is replicated to some other layers.

**Comment by andreasneumann** [comment]
So you want to copy widget configurations within the same layer - not to another layer - right?

**Comment by ponceta** [comment]
> So you want to copy widget configurations within the same layer - not to another layer - right?

Yes!

**Comment by 3nids** [comment]
You can already to this by selecting the editor widget category from copy/paste style. This would reuse what is possible (fields with the same names)

**Comment by andreasneumann** [comment]
> You can already to this by selecting the editor widget category from copy/paste style. This would reuse what is possible (fields with the same names)

@3nids - I believe this is a different thing. This would be to copy widget configs from one layer to another, but if I understand it correctly, @ponceta wants to copy widget configuration from one field to another within the same layer.

**Comment by gacarrillor** [comment]
@ponceta, nice feature request indeed!

Just a couple of questions: 

1. The copy&paste context menu should be displayed only for items of the `Fields` category in the `Available Widgets` panel or do you expect it also for items of the `Relations` category?

2. Do you imagine a _scope_ selection between the copy and paste actions (e.g., a dialog to select which scopes you want to copy) or perhaps something more direct (i.e., copy all the scopes you listed and paste what makes sense to paste)?

**Comment by gacarrillor** [comment]
We probably want to avoid copying field comments as well?

**Comment by ponceta** [comment]
1. This is a different copy paste but yes we could extend it to copy pasting relations widget configuration. But the copy paste will be available from relations to relations.
2. If we distinct scope we could copy either parts of the field widget configuration or all sections:
- **ALL**
- affichage, 
- general, 
- type, 
- constraints, 
- defaults, 
- strategies

**Comment by gacarrillor** [comment]
@ponceta, would copy from a field and paste to the very same field be useful for your use cases?

At first we thought it would be disabled, but it also might make sense to copy a field's config to keep it in the clipboard, do some changes, visualize and test those changes, and after that, return to what we had in the clipboard (i.e., paste the config in the same field as a way to undo things).

What do you think?

**Comment by ponceta** [comment]
> @ponceta, would copy from a field and paste to the very same field be useful for your use cases?

Yes, between two projects would be the most probable scenario but I see no strong necessity of disabling it.
